src/Scripting/CSharpTest/ScriptTests.cs

```diff
@@ -178,6 +178,75 @@ public void Do()
 , ScriptOptions.Default.WithReferences(MscorlibRef, SystemRef, SystemCoreRef, CSharpRef));
         }
 
+        [WorkItem(6676, "https://github.com/dotnet/roslyn/issues/6676")]
+        [Fact]
+        public void TestRunEmbeddedStatementNotFollowedBySemicolon()
+        {
+            var exceptionThrown = false;
+
+            try
+            {
+                var state = CSharpScript.RunAsync(@"
+ if (true)
+ System.Console.WriteLine(true)", globals: new ScriptTests());
+            }
+            catch (CompilationErrorException ex)
+            {
+                exceptionThrown = true;
+                ex.Diagnostics.Verify(
+                // (3,32): error CS1002: ; expected
+                //  System.Console.WriteLine(true)
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(3, 32));
+            }
+
+             Assert.True(exceptionThrown);
+        }
+
+        [WorkItem(6676, "https://github.com/dotnet/roslyn/issues/6676")]
+        [Fact]
+        public void TestRunEmbeddedStatementFollowedBySemicolon()
+        {
+            var state = CSharpScript.RunAsync(@"
+if (true)
+System.Console.WriteLine(true);", globals: new ScriptTests());
+            Assert.Null(state.Exception);
+        }
+
+        [WorkItem(6676, "https://github.com/dotnet/roslyn/issues/6676")]
+        [Fact]
+        public void TestRunStatementFollowedBySpace()
+        {
+            var state = CSharpScript.RunAsync(@"
+System.Console.WriteLine(true) ", globals: new ScriptTests());
+            Assert.Null(state.Exception);
+        }
+
+        [WorkItem(6676, "https://github.com/dotnet/roslyn/issues/6676")]
+        [Fact]
+        public void TestRunStatementFollowedByNewLineNoSemicolon()
+        {
+            var exceptionThrown = false;
+
+            try
+            {
+                var state = CSharpScript.RunAsync(@"
+System.Console.WriteLine(true)\n", globals: new ScriptTests());
```

Comment:
I see why the ```\n``` is causing a problem. The string is a verbatim string (the ```@``` prefix is used), therefore we actually have two separate characters ```\``` and ```n``` at the end rather than a line break. Given that, we probably don't need this test after all.